### PR TITLE
bpo-43762: Add audit events for loading of sqlite3 extensions

### PR DESCRIPTION
**Pull request by erlend-aasland**
https://bugs.python.org/issue43762

### CODE REVIEW

**Comment by erlend-aasland** [comment]
cc. @tiran

**Comment by erlend-aasland** [comment]
@tiran & @zooba, can one of you review this? FYI, the previous `sqlite3` audit events were added in #14301 by Steve.

File: Lib/test/audit-tests.py [hunk 1]
```diff
@@ -341,6 +341,29 @@ def hook(event, args):
...
+    def hook(event, *args):
+        if event.startswith("sqlite3."):
+            print(event, *args)
+
+    sys.addaudithook(hook)
+    cx = sqlite3.connect(":memory:")
+
+    # Configured without --enable-loadable-sqlite-extensions
+    if not hasattr(sqlite3.Connection, "enable_load_extension"):
+        return
```
Comment (zooba) [inline Lib/test/audit-tests.py]: Can we swap the logic on this to contain the below code in the `if` block? That way if someone comes to add more steps later on, it's more obvious that they can be added to this test even if they don't depend on the extension loading.

File: Modules/_sqlite/connection.c [hunk 2]
```diff
@@ -1145,6 +1145,10 @@ pysqlite_connection_enable_load_extension_impl(pysqlite_Connection *self,
 {
     int rc;
 
+    if (PySys_Audit("sqlite3.enable_load_extension", "i", onoff) < 0) {
```
Comment (zooba) [inline Modules/_sqlite/connection.c]:
```suggestion
    if (PySys_Audit("sqlite3.enable_load_extension", "O", onoff ? Py_True : Py_False) < 0) {
```

**Review by zooba — COMMENTED** [review]
Couple of minor tweaks, mainly for readability/ease of use.

I wonder if it's worth returning the connection object when it's created (through a new event in module.c) and then reference it in these events? That can then correlate these (and other) events with the file - we do this already for sockets.

After some thought, I think it's probably not worth it for these ones. The important information is in the extension being loaded, and it doesn't really relate to the connection at all. However, if we wanted to add it later, we couldn't. So might be worth doing now?

File: Lib/test/audit-tests.py (hunk 1 above)
Comment (erlend-aasland) [reply Lib/test/audit-tests.py]: Good point. Thanks!

**Comment by erlend-aasland** [comment]
FYI, rebased onto `main` to resolve conflicts.

File: Doc/library/sqlite3.rst [hunk 3]
```diff
@@ -467,8 +467,13 @@ Connection Objects
 
       Loadable extensions are disabled by default. See [#f1]_.
 
+      .. audit-event:: sqlite3.enable_load_extension enabled sqlite3.enable_load_extension
```
Comment (zooba) [inline Doc/library/sqlite3.rst]:
```suggestion
      .. audit-event:: sqlite3.enable_load_extension connection,enabled sqlite3.enable_load_extension
```

File: Doc/library/sqlite3.rst [hunk 4]
```diff
@@ -479,8 +484,13 @@ Connection Objects
 
       Loadable extensions are disabled by default. See [#f1]_.
 
+      .. audit-event:: sqlite3.load_extension path sqlite3.load_extension
```
Comment (zooba) [inline Doc/library/sqlite3.rst]:
```suggestion
      .. audit-event:: sqlite3.load_extension connection,path sqlite3.load_extension
```

**Review by zooba — APPROVED** [review]
Approved with the doc update suggestions

File: Doc/library/sqlite3.rst (hunk 3 above)
Comment (erlend-aasland) [reply Doc/library/sqlite3.rst]: Thanks!

**Comment by erlend-aasland** [comment]
> Approved with the doc update suggestions

Thanks for reviewing! BTW, I used `/handle` iso. `/result` (ctypes uses that already). I figured it aligned better with the SQLite jargon.

**Pull request merged**